.github/workflows/node-ci.yml

```diff
@@ -140,43 +140,8 @@ jobs:
             libc++abi-9-dev
           /usr/sbin/update-ccache-symlinks
 
-      - name: npm ci
-        run: npm ci --ignore-scripts
-
-      - name: Prepare ccache
-        run: ccache --clear --set-config cache_dir=~/.ccache
-
-      - name: Cache ccache
-        uses: actions/cache@v3
-        env:
-          cache-name: ccache-v1
-        with:
-          path: ~/.ccache
-          key: ${{ env.cache-name }}-${{ matrix.os }}-${{ github.job }}-${{ github.ref }}-${{ github.sha }}-${{ github.head_ref }}
-          restore-keys: |
-            ${{ env.cache-name }}-${{ matrix.os }}-${{ github.job }}-${{ github.ref }}-${{ github.sha }}
-            ${{ env.cache-name }}-${{ matrix.os }}-${{ github.job }}-${{ github.ref }}
-            ${{ env.cache-name }}-${{ matrix.os }}-${{ github.job }}
-
-      - name: Clear ccache statistics
-        run: |
-          ccache --zero-stats --set-config cache_dir=~/.ccache
-          ccache --max-size=2G --set-config cache_dir=~/.ccache
-          ccache --show-stats --set-config cache_dir=~/.ccache
-
-      - name: CMake
-        if: runner.os == 'MacOS'
-        run: |
-          cmake . -B build -G Ninja -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -DCMAKE_BUILD_TYPE=${{ env.BUILDTYPE }}
-
-      - name: CMake
-        if: runner.os == 'Linux'
-        run: |
-          cmake . -B build -G Ninja -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -DCMAKE_BUILD_TYPE=${{ env.BUILDTYPE }} -DCMAKE_C_COMPILER=gcc-10 -DCMAKE_CXX_COMPILER=g++-10
-
-      - name: Build
-        run: |
-          cmake --build build -j $(nproc 2>/dev/null || sysctl -n hw.ncpu 2>/dev/null)
+      - name: npm build
```

Comment:
Is `ccache` supported by `npm build` (did it actually work now)?

Reply:
I'd say right now this is not using ccache, the cmake-js option it is using looks like this

```
[
  '/root/.nvm/versions/node/v16.15.0/bin/node',
  '/opt/test7/cmake/node_modules/.bin/cmake-js',
  'compile'
]
info TOOL Using Ninja generator, because ninja is available.
info CMD CONFIGURE
info RUN [
info RUN   'cmake',
info RUN   '/opt/test7/cmake',
info RUN   '--no-warn-unused-cli',
info RUN   '-G',
info RUN   'Ninja',
info RUN   '-DCMAKE_JS_VERSION=undefined',
info RUN   '-DCMAKE_BUILD_TYPE=Release',
info RUN   '-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=/opt/test7/cmake/build/Release',
info RUN   '-DCMAKE_MSVC_RUNTIME_LIBRARY=MultiThreaded$<$<CONFIG:Debug>:Debug>',
info RUN   '-DCMAKE_JS_INC=/root/.cmake-js/node-x64/v16.15.0/include/node',
info RUN   '-DCMAKE_JS_SRC=',
info RUN   '-DNODE_RUNTIME=node',
info RUN   '-DNODE_RUNTIMEVERSION=16.15.0',
info RUN   '-DNODE_ARCH=x64',
info RUN   '-DCMAKE_CXX_FLAGS=-DBUILDING_NODE_EXTENSION'
info RUN ]
```

I think I could add the ccache commands back to the workflow and set the option to use it in cmake with 
`npm config set cmake_CMAKE_CXX_COMPILER_LAUNCHER ccache`

which adds -DCMAKE_CXX_COMPILER_LAUNCHER=ccache to the options

Reply:
well i made the changes to include that build option, and can see it is there now, but beyond that I'm not sure how I tell it is being used.

not sure why the macos ci is failing...i don't think it is related to this PR.

Reply:
@ntadej @acalcutt CI looks like it is passing now. Should we mark this as resolved?